java/gateway-core/src/main/java/com/avaulta/gateway/pseudonyms/impl/Sha256PseudonymEncoder.java

```diff
@@ -0,0 +1,43 @@
+package com.avaulta.gateway.pseudonyms.impl;
+
+import com.avaulta.gateway.pseudonyms.Pseudonym;
+import com.avaulta.gateway.pseudonyms.PseudonymEncoder;
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * implementation of defacto encoding used by BulkDataSanitizerImpl as of v0.4.30
+ */
+public class Sha256PseudonymEncoder implements PseudonymEncoder {
+
+
+    @Override
+    public String encode(Pseudonym pseudonym) {
+        return base64Encode(pseudonym.getHash());
+    }
+
+    @Override
+    public Pseudonym decode(String input) {
+        return Pseudonym.builder()
+            .hash(input.getBytes(StandardCharsets.UTF_8))
+            .build();
+    }
+
+    @Override
+    public boolean canBeDecoded(String possiblePseudonym) {
+        return possiblePseudonym != null &&
+            possiblePseudonym.getBytes(StandardCharsets.UTF_8).length == 43; //43 rather than 32, bc of base64 encoding
```

Comment:
```suggestion
            possiblePseudonym.getBytes(StandardCharsets.UTF_8).length == 43; //43 rather than 32, bc of base64 encoding without padding
```

Reply:
could also go a bit further and try to check that input contains valid base64 characters, but I think there's no easy way of doing it with the jre library (rather than try to decode and catch)